packages/smooth_app/lib/query/products_preload_helper.dart

```diff
@@ -1,23 +1,26 @@
 import 'dart:io';
 import 'package:openfoodfacts/openfoodfacts.dart';
 import 'package:smooth_app/database/dao_product.dart';
-import 'package:smooth_app/database/local_database.dart';
 import 'package:smooth_app/query/product_query.dart';
 
 class PreloadDataHelper {
-  PreloadDataHelper(this.localDatabase);
-  LocalDatabase localDatabase;
-
-  Future<String> preloadData() async {
-    final DaoProduct daoProduct = DaoProduct(localDatabase);
+  PreloadDataHelper(this.daoProduct);
+  DaoProduct daoProduct;
+  Future<String> getTopProducts() async {
+    List<String> allProductCodes = await daoProduct.getAllKeys();
+    final Map<String, Product> allProducts =
+        await daoProduct.getAll(allProductCodes);
+    allProducts.removeWhere(
+        (String key, Product value) => value.knowledgePanels == null);
+    allProductCodes = allProducts.keys.toList();
     final List<ProductField> fields = ProductQuery.fields;
     fields.remove(ProductField.KNOWLEDGE_PANELS);
     try {
       final ProductSearchQueryConfiguration queryConfig =
           ProductSearchQueryConfiguration(
         fields: fields,
         parametersList: <Parameter>[
-          const PageSize(size: 1000),
+          const PageSize(size: 100),
```

Comment:
Magic number: please put that in a `const`, and possibly make it explicit in the button text "Loading the top 100 products".

Reply:
My bad, I just wanted to relive the server when  I was testing, forgot to edit that later